### DISCUSSION TITLE
Use of constrain(..., -outMax, outMax) -> shouldn't this be constraint(..., outMin, outMax) ?

### DISCUSSION THREAD

**Discussion by JonNelson** [Dlloydev/QuickPID#60] [Q&A]
https://github.com/Dlloydev/QuickPID/blob/d0611bc797d5d6c916809773397de968e3ff9fe6/src/QuickPID.cpp#L100

**Comment by Dlloydev** [comment]
The purpose of using `-outMax` is to allow unwinding the integral accumulator by an amount that's limited equally for [both positive and negative directions](https://info.erdosmiller.com/blog/pid-anti-windup-techniques) when the controller output is saturated. The anti-windup code was inspired from [here](https://www.embeddedrelated.com/showcode/346.php).

Test results:

![image](https://user-images.githubusercontent.com/63488701/210589171-26d1026d-360e-4893-9764-d6074fd67e52.png)
